rubyforgood/diaper
# inventory updating problem #363

Resolves #363

### Description

This fix addresses issue #363 but I don't think it's a very robust solution. A better solution would touch more code though and I'll follow up with an issue to discuss that approach.



### Type of change



* Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

- updated specs relevant to this method
- given that this method is only called when updating a donation or purchase some of the specs testing this method did not seem to test relevant behavior. I removed those methods please double check to make sure that makes sense.

Besides updating the specs, to test manually:

- Go to http://localhost:3000/pdx_bank/items
- click on the Items, Quantity and Location Tab
- pick an item that has a total quantity present (PDX Diaper Bank Office from seed data has multiple line items with data)

Scenario 1
- open another tab and navigate to donations
- create a donation where you specify a quantity
- make a correction by changing the quantity of the donation 
- go back to the tab listing the items and refresh
- navigate to the Quantity and Location Tab
- make sure the total quantity is changed by the specified quantity in the "make a correction step"

Scenario 2
- repeat the steps from Scenario 1 for a purchase instead of a donation

### Screenshots
<img width="1459" alt="1" src="https://user-images.githubusercontent.com/1443346/41185631-4c92e498-6b58-11e8-964e-bfdd0e5b06e3.png">
<img width="1437" alt="2" src="https://user-images.githubusercontent.com/1443346/41185639-59f49e2e-6b58-11e8-844b-ac9b57566b24.png">
<img width="1440" alt="3" src="https://user-images.githubusercontent.com/1443346/41185642-5c554204-6b58-11e8-95b4-3af855afbd93.png">
